pkg/fx/pdp/provider.go

```diff
@@ -0,0 +1,103 @@
+package pdp
+
+import (
+	"go.uber.org/fx"
+	"gorm.io/gorm"
+
+	"github.com/storacha/piri/pkg/config/app"
+	echofx "github.com/storacha/piri/pkg/fx/echo"
+	"github.com/storacha/piri/pkg/pdp"
+	"github.com/storacha/piri/pkg/pdp/aggregator"
+	"github.com/storacha/piri/pkg/pdp/chainsched"
+	"github.com/storacha/piri/pkg/pdp/ethereum"
+	"github.com/storacha/piri/pkg/pdp/httpapi/server"
+	"github.com/storacha/piri/pkg/pdp/pieceadder"
+	"github.com/storacha/piri/pkg/pdp/piecefinder"
+	"github.com/storacha/piri/pkg/pdp/scheduler"
+	"github.com/storacha/piri/pkg/pdp/service"
+	"github.com/storacha/piri/pkg/pdp/types"
+	"github.com/storacha/piri/pkg/store/blobstore"
+	"github.com/storacha/piri/pkg/store/stashstore"
+)
+
+var Module = fx.Module("pdp-service",
+	fx.Provide(
+		fx.Annotate(
+			ProvidePDPService,
+			fx.As(fx.Self()),      // provide service as concrete type
+			fx.As(new(types.API)), // also provide the server as the interface(s) it implements
+			fx.As(new(types.ProofSetAPI)),
+			fx.As(new(types.PieceAPI)),
+		),
+		fx.Annotate(
+			ProvideProofSetIDProvider,
+			fx.As(new(aggregator.ProofSetIDProvider)),
+		),
+
+		fx.Annotate(
+			ProvidePoorlyNamedPDPInterface,
+			fx.As(new(pdp.PDP)),
+		),
+		fx.Annotate(
+			server.NewPDPHandler,
+			fx.As(new(echofx.RouteRegistrar)),
+			fx.ResultTags(`group:"route_registrar"`),
+		),
+	),
+)
+
+type Shit struct {
+	aggregator  aggregator.Aggregator
+	pieceFinder piecefinder.PieceFinder
+	pieceAdder  pieceadder.PieceAdder
+}
+
+func (s *Shit) PieceAdder() pieceadder.PieceAdder {
+	return s.pieceAdder
+}
+
+func (s *Shit) PieceFinder() piecefinder.PieceFinder {
+	return s.pieceFinder
+}
+
+func (s *Shit) Aggregator() aggregator.Aggregator {
+	return s.aggregator
+}
+
+func ProvidePoorlyNamedPDPInterface(service types.API, agg aggregator.Aggregator, cfg app.AppConfig) (*Shit, error) {
```

Comment:
OMG. man this is painful though I get it. look I don't want to make a big deal but sir this is a wendys. can we not use the actual S-word in the code? Just in case we forget to change it?

Reply:
🙈 sorry!!!! I'll fix this, didn't mean to push it.